tests/rgen/Microsoft.Macios.Generator.Tests/Extensions/NamedTypeSymbolExtensionsTests.cs

```diff
@@ -0,0 +1,110 @@
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Macios.Generator.Extensions;
+using Xamarin.Tests;
+using Xamarin.Utils;
+using Xunit;
+
+namespace Microsoft.Macios.Generator.Tests.Extensions;
+
+public class NamedTypeSymbolExtensionsTests : BaseGeneratorTestClass {
+
+	[Theory]
+	[AllSupportedPlatforms]
+	public void TryGetEnumFieldsNotEnum (ApplePlatform platform)
+	{
+		const string inputString = @"
+namespace Test;
+public class NotEnum {
+}
+";
+		var (compilation, syntaxTrees) = CreateCompilation (nameof (TryGetEnumFieldsNotEnum), platform, inputString);
+		Assert.Single (syntaxTrees);
+		var declaration = syntaxTrees [0].GetRoot ()
+			.DescendantNodes ()
+			.OfType<BaseTypeDeclarationSyntax> ()
+			.FirstOrDefault ();
+		Assert.NotNull (declaration);
+		var semanticModel = compilation.GetSemanticModel (syntaxTrees [0]);
+		var symbol = semanticModel.GetDeclaredSymbol (declaration);
+		Assert.NotNull (symbol);
+		Assert.False (symbol.TryGetEnumFields (out var fields, out var diagnostics));
+		Assert.Null (fields);
+		Assert.Single (diagnostics);
+	}
+
+	const string emptyEnum = @"
+namespace Test;
+public enum MyEnum {
+}
+";
+
+	const string missingFieldAttributes = @"
+namespace Test;
+public enum MyEnum {
+	First,
+	Second,
+	Last,
+}
+";
+
+	[Theory]
+	[AllSupportedPlatforms (emptyEnum)]
+	[AllSupportedPlatforms (missingFieldAttributes)]
+	public void TryGetEnumFieldsNoFields (ApplePlatform platform, string inputString)
+	{
+		var (compilation, syntaxTrees) = CreateCompilation (nameof (TryGetEnumFieldsNotEnum), platform, inputString);
+		Assert.Single (syntaxTrees);
+		var declaration = syntaxTrees [0].GetRoot ()
+			.DescendantNodes ()
+			.OfType<BaseTypeDeclarationSyntax> ()
+			.FirstOrDefault ();
+		Assert.NotNull (declaration);
+		var semanticModel = compilation.GetSemanticModel (syntaxTrees [0]);
+		var symbol = semanticModel.GetDeclaredSymbol (declaration);
+		Assert.NotNull (symbol);
+		Assert.True (symbol.TryGetEnumFields (out var fields, out var diagnostics));
+		Assert.NotNull (fields);
+		Assert.Empty (fields);
+		Assert.Null (diagnostics);
+	}
+
+	[Theory]
+	[AllSupportedPlatforms]
+	public void TryGetEnumFieldsWithAttr (ApplePlatform platform)
+	{
+		const string inputString = @"
+using ObjCRuntime;
+using ObjCBindings;
+
+namespace Test;
+public enum MyEnum {
+	[Field<EnumValue> (""FirstBackendField"")]
+	First,
+	[Field<EnumValue> (""SecondBackendField"")]
```

Comment:
Another scenario, leading or trailing whitespace:
```suggestion
	[Field<EnumValue> ("" \tSecondBackendField\t \n"")]
```

Reply:
Wrong test, I'll add that in the correct one that uses the TestDataTryGetEnumFieldsNoFields data class.

Reply:
Added in  https://github.com/xamarin/xamarin-macios/pull/21562/commits/c0270b78dd1f2506c015c889a2dc8cb03787561b